bin.src/plot_jointcal_results.py

```diff
@@ -0,0 +1,105 @@
+"""
+Generate plots from the results of a jointcal run.
+
+Requires a butler repository that has had jointcal successfully run on it, which
+produces a new "wcs" component of the repository.
+"""
+
+import os
+
+import lsst.daf.persistence
+from lsst.meas.astrom import LoadAstrometryNetObjectsTask, LoadAstrometryNetObjectsConfig
+import lsst.afw.image
+from lsst.afw.geom import degrees
+
+from lsst.jointcal import utils
+
+
+def get_valid_dataIds(butler, dataset_type='wcs'):
+    """Return a list of all dataIds that exist in this butler."""
+    data_ids = []
+    for data_ref in butler.subset(dataset_type):
+        if data_ref.datasetExists():
+            data_ids.append(data_ref.dataId)
+    return data_ids
+
+
+def prep_reference_loader(center, radius):
+    """
+    Return an astrometry.net reference loader.
+
+    Parameters
+    ----------
+    center: afw.coord
+        The center of the field you're testing on.
+    radius: afw.geom.angle
+        The radius to load objects around center.
+    """
+    refLoader = LoadAstrometryNetObjectsTask(LoadAstrometryNetObjectsConfig())
+    # Make a copy of the reference catalog for in-memory contiguity.
+    return refLoader.loadSkyCircle(center, radius, filterName='r').refCat.copy()
+
+
+def get_old_wcs_list(data_refs):
+    """Return a list of the original WCSs of the data_refs."""
+    result = []
+    for data_ref in data_refs:
+        md = data_ref.get("calexp_md", immediate=True)
+        tanwcs = lsst.afw.image.TanWcs.cast(lsst.afw.image.makeWcs(md))
+        result.append(tanwcs)
+    return result
+
+
+def get_final_path(path):
+    """Return the final part of a path, ignoring a trailing `/`."""
+    name = os.path.split(path)
+    if name[1] == '':
+        name = os.path.split(name[0])[1]
+    else:
+        name = name[1]
+    return name
+
+
+def main():
+    import argparse
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("repo", metavar="repo", help="Directory containing butler repository.")
+    parser.add_argument("refcat", metavar="refcat", help="Directory of reference catalog to match against.")
+    parser.add_argument("ra", default=None, type=float, nargs='?',
+                        help="Decimal degrees center RA of catalog (default uses center of first visit).")
+    parser.add_argument("dec", default=None, type=float, nargs='?',
+                        help="Decimal degrees center Dec of catalog (default uses center of first visit).")
+    parser.add_argument("radius", default=6., type=float, nargs='?',
+                        help="Radius (degrees) of sources to load from reference catalog.")
+    parser.add_argument("-i", "--interactive", action="store_true",
+                        help="Use interactive matplotlib backend and set ion(), in addition to saving files.")
+    args = parser.parse_args()
```

Comment:
It may be useful here to include an argument that allows the user to specify an output directory